core/src/test/scala/org/apache/spark/scheduler/DAGSchedulerSuite.scala

```diff
@@ -598,6 +598,88 @@ class DAGSchedulerSuite
     assertDataStructuresEmpty()
   }
 
+  test("run with ShuffleMapStage retry") {
```

Comment:
sorry I should have suggested this earlier, can we also make this name more specific?  maybe "no failures when map output made available by zombie task attempt" ?